### DISCUSSION TITLE
How to update ByteRange once Signature added

### DISCUSSION THREAD

**Discussion by ismaelwenyam** [pdfcpu/pdfcpu#966] [Q&A]
Hi folks,
i am using pdfcpu in order to add a signature lin these steps :

`       sigDict := types.Dict(map[string]types.Object{
		"ByteRange": types.NewIntegerArray(0, 0, 0, 0),
		"Type":      types.Name("Sig"),
		"Filter":    types.Name("Adobe.PPKLite"),
		"SubFilter": types.Name("ETSI.CAdES.detached"),
		"Contents":  types.NewHexLiteral(signatureData),
		"M":         types.StringLiteral(types.DateString(now)),
	})
ir, err := ctx.XRefTable.IndRefForNewObject(sigDict)
	if err != nil {
		return err
	}

	sigFieldDict := types.Dict(map[string]types.Object{
		"Type":    types.Name("Annot"),
		"Subtype": types.Name("Widget"),
		"FT":      types.Name("Sig"),
		"T":       types.StringLiteral("Signature"),
		"Rect":    types.NewIntegerArray(0, 0, 0, 0),
		"V":       *ir,
	})

	if ir, err = ctx.XRefTable.IndRefForNewObject(sigFieldDict); err != nil {
		return err
	}
	pg, _ := ctx.XRefTable.DereferenceDictEntry(ctx.XRefTable.RootDict, "Pages")
	pgd := pg.(types.Dict)
	kids := pgd.ArrayEntry("Kids")
	p0, _ := ctx.XRefTable.DereferenceDict(kids[0])

	annots := types.Array{*ir}
	p0.Update("Annots", annots)

	acroFormDict := types.Dict(map[string]types.Object{
		"Fields":   types.Array{*ir},
		"SigFlags": types.Integer(3),
	})

	if ir, err = ctx.XRefTable.IndRefForNewObject(acroFormDict); err != nil {
		return err
	}

	rootDic, _ := ctx.XRefTable.Catalog()
	rootDic.Insert("AcroForm", *ir)`
then after that i write the context to file, i subsequently calculate the correct ByteRange values to cover the entire pdf minus the signature itself.
Now the problem i am facing is how to patch the ByteRange without corrupting the pdf.
Does pdfcpu supports this king of situation?
Thank you in advance!

**Comment by hhrutter** [comment]
Digital signature handling is an open issue and being worked on.
Please check back #168.

**Comment by ismaelwenyam** [comment]
thank you @hhrutter.
